src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,170 @@
+   This automatically enables built-in state restoration
+   for those widgets.
+
+2. For custom widgets,
+   you must decide what state you want to restore
+   and hold that state in a [`RestorableProperty`][].
+   (The Flutter API provides various subclasses for
+   different data types.)
+   Define those `RestorableProperty` widgets 
+   in a `State` class that uses the [`RestorationMixin`][].
+   Register those widgets with the mixin in a
+   `restoreState` method.
+
+3. If you use any Navigator API (like `push`, `pushNamed`, and so on)
+   migrate to the API that has "restorable" in the name
+   (`restorablePush`, `resstorablePushNamed`, and so on)
+   to restore the navigation stack.
+
+Other considerations:
+
+* Providing a `restorationId` to
+  `MaterialApp`, `CupertinoApp`, or `WidgetsApp`
+  automatically enables state restoration by
+  injecting a `RootRestorationScope`.
+  If you need to restore state _above_ the app class,
+  inject a `RootRestorationScope` manually.
+
+* **The difference between a `restorationId` and
+  a `restorationScopeId`:** Widgets that take a
+  `restorationScopeID` create a new `restorationScope`
+  (a new `RestorationBucket`) into which all children
+  store their state. A `restorationId` means the widget
+  (and its children) store the data in the surrounding bucket.
+
+[a bit of extra setup]: {{site.api}}/flutter/services/RestorationManager-class.html#state-restoration-on-ios
+[`restorationId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationId.html
+[`restorationScopeId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationScopeId.html
+[`RestorationMixin`]: {{site.api}}/flutter/widgets/RestorationMixin-mixin.html
+[`RestorationScope`]: {{site.api}}/flutter/widgets/RestorationScope-class.html
+[`restoreState`]: {{site.api}}/flutter/widgets/RestorationMixin/restoreState.html
+[VeggieSeasons]: {{site.github}}/flutter/samples/tree/master/veggieseasons
+
+## Restoring navigation state
+
+If you want your app to return to a particular route
+that the user was most recently viewing
+(the shopping cart, for example), then you must implement
+restoration state for navigation, as well.
+
+If you use the Navigator API directly,
+migrate the standard methods to restorable
+methods (that have "restorable" in the name).
+For example, replace `push` with [`restorablePush`][].
+
+The VeggieSeasons example (listed under "Other resources" below)
+implements navigation with the [`go_router`][] package.
+Setting the `restorationId`
+values occur in the `lib/screens` classes.
+
+## Testing state restoration
+
+To test state restoration, set up your mobile device so that
+it doesn't save state once an app is backgrounded.
+To learn how to do this for both iOS and Android,
+check out [Testing state restoration][] on the
+[`RestorationManager`][] page.
+
+{{site.alert.warning}}
+  **Don't forget to reenable
+  storing state on your device once you are
+  finished with testing!**
```

Comment:
Consider not making this bold, since **Warning** already is. At least for me, it makes it easier to read.

```suggestion
  Don't forget to reenable
  storing state on your device once you are
  finished with testing!
```